yt_dlp/YoutubeDL.py

```diff
@@ -611,22 +611,23 @@ def __init__(self, params=None, auto_init=True):
 
         if self.params.get('no_color'):
             if self.params.get('color') is not None:
-                self.report_warning('Using')
+                self.report_warning('Overwriting params from "color" with "no_color"')
             self.params['color'] = 'no_color'
 
         term_allow_color = os.environ.get('TERM', '').lower() != 'dumb'
 
         def process_color_policy(stream):
-            stream_name = {sys.stdout: 'stdout', sys.stderr: 'stderr'}.get(stream)
-            policy = traverse_obj(self.params, ('color', (stream_name, None)), get_all=False)
-            if policy not in (True, False, 'no_color'):
+            stream_name = {sys.stdout: 'stdout', sys.stderr: 'stderr'}[stream]
+            policy = traverse_obj(self.params, ('color', (stream_name, None), {str}), get_all=False)
+            if policy in ('auto', None):
                 return term_allow_color and supports_terminal_sequences(stream)
-            return policy
+            return {'always': True, 'never': False}.get(policy, policy)
```

Comment:
Pls add back the assert. We don't want to support arbitrary values. Also, I don't think `{str}` is needed.

Reply:
`{str}` is needed bc if we have a `None` value it will search the second path, which will be a dict, giving the `dict is unhashable` error. `assert` has to be added though